lib/macho/load_commands.rb

```diff
@@ -399,6 +339,18 @@ def initialize(view, cmd, cmdsize, name, timestamp, current_version, compatibili
       @current_version = current_version
       @compatibility_version = compatibility_version
     end
+
+    # @param endianness [Symbol, nil] the endianness of the serialized string,
+    #  or nil to use the view's endianness
+    # @return [String] the serialized fields of the load command
+    # @api private
+    def serialize(endianness = nil)
+      endianness ||= view.endianness
+      format = Utils.specialize_format(FORMAT, endianness)
+      padded_name = name.to_s + "\x00" * name.npad
+      [cmd, cmdsize, name.to_i, timestamp, current_version,
+        compatibility_version].pack(format) + padded_name
```

Comment:
One problem this currently has is that `name.to_i` (`name#@string_offset`) is currently `0` and isn't updated, another is that the padding isn't adjusted to the bit width of the target Mach-O file this will be serialized into.

Here's a suggestion for how this part of the serialization could work (and avoid hard-coding the padding logic into every `serialize` method that has to deal with `LCStr`s):

```ruby
string_payload, string_offsets = Utils.pack_strings(SIZEOF, target.magic32?, :name => name.to_s)
cmdsize = SIZEOF + string_payload.bytesize
[cmd, cmdsize, string_offsets[:name], timestamp, current_version,
  compatibility_version].pack(format) + string_payload
```

This means all the details of computing the offset where a string starts, packing (possibly multiple) strings into a byte stream, then padding this byte stream with zeros to make the load command meet alignment requirements would all be done in the utility function. This avoids encoding too many details into every single `serialize` method. The implementation could look something like (untested):

```ruby
def self.pack_strings(fixed_size, is_32bit, strings = {})
  offsets = {}
  next_offset = fixed_size
  payload = ""
  strings.each do |key, string|
    offsets[key] = next_offset
    payload << string
    payload << "\x00"
    next_offset += string.bytesize + 1
  end
  payload << "\x00" * compute_required_padding(fixed_size + payload.bytesize, is_32bit)
  [payload, offsets]
end
```